### DISCUSSION TITLE
Add a toggle to remove the third button from the key overlay

### DISCUSSION THREAD

**Discussion by Flamiii** [ppy/osu#25990] [Ideas]
A lot of people (myself included) don't ever use the smoke key, so having the option to remove the third button from the key overlay in the skin editor would be nice.

Edit: I see https://github.com/ppy/osu/discussions/20625 but the HUD redesign for gameplay already kind of happened and the key overlay is movable. You still can't hide the third key.

**Comment by peppy** [comment]
Can be added as a setting in the skin editor, but requires some tooling to mark keys as auxiliary etc.

**Reply by NY03M** [reply]
I was looking into this recently while creating a minimalist legacy skin and ended up tracing the current implementation through LegacyKeyCounter, KeyCounterActionTrigger<T> and RulesetInputManager<T>.

From what I found, the legacy input overlay creates a counter for every OsuAction, which is why the smoke action is always displayed as B3. Because of that there doesn't currently seem to be a way for skins to hide only the smoke counter without also hiding the other labels.

I just wanted to mention that this is still an issue today for transparent/minimalist legacy overlays, and I'd still find the proposed auxiliary-key approach useful if possible :).

I know it might sound like something that shouldn't be an issue but it really does get in the way sometimes for some overlays...
